### PR TITLE
raftstore: schedule log gc by size

### PR DESCRIPTION
**Pull request by BusyJay**
ref #1337 

@siddontang @hhkbp2 @zhangjinpeng1987 PTAL

### CODE REVIEW

File: src/raftstore/store/config.rs [hunk 1]
```diff
@@ -21,9 +21,10 @@ const RAFT_HEARTBEAT_TICKS: usize = 10;
 const RAFT_ELECTION_TIMEOUT_TICKS: usize = 50;
 const RAFT_MAX_SIZE_PER_MSG: u64 = 1024 * 1024;
 const RAFT_MAX_INFLIGHT_MSGS: usize = 256;
-const RAFT_LOG_GC_INTERVAL: u64 = 5000;
+const RAFT_LOG_GC_INTERVAL: u64 = 10000;
 const RAFT_LOG_GC_THRESHOLD: u64 = 50;
-const RAFT_LOG_GC_LIMIT: u64 = 100000;
+const RAFT_LOG_GC_COUNT_LIMIT: u64 = 100000;
```
Comment (siddontang) [inline src/raftstore/store/config.rs]: need 100000 now?

File: src/raftstore/store/store.rs [hunk 2]
```diff
@@ -1159,7 +1160,10 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             let applied_idx = peer.get_store().applied_index();
             let first_idx = peer.get_store().first_index();
             let compact_idx;
-            if applied_idx > first_idx && applied_idx - first_idx >= self.cfg.raft_log_gc_limit {
+            if applied_idx > first_idx &&
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: add log and metric for exceeding limit.

**Comment by siddontang** [comment]
Rest LGTM

File: src/raftstore/store/store.rs (hunk 2 above)
Comment (BusyJay) [inline src/raftstore/store/store.rs]: This is tracked by CompactLog command already.

**Comment by BusyJay** [comment]
Ping

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by BusyJay** [comment]
@siddontang PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**